src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,171 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.global.concurrent.TracedCallable;
+import io.opentracing.contrib.global.concurrent.TracedRunnable;
+import io.opentracing.propagation.Format;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The <code>GlobalTracer</code> forwards all methods to a single {@link Tracer} implementation that can be
+ * instantiated in one of two ways:
+ * <ol>
+ * <li>Explicitly, by calling {@link #register(Tracer)} with a configured tracer implementation, or:</li>
+ * <li>Automatically by using the Java <code>ServiceLoader</code> SPI mechanism to load an implementation
+ * from the classpath.</li>
+ * </ol>
+ *
+ * @author Sjoerd Talsma
+ * @navassoc - provides 1 io.opentracing.Tracer
+ * @navassoc - uses - io.opentracing.contrib.activespan.ActiveSpanManager
+ * @see Tracer
+ */
+public final class GlobalTracer implements Tracer {
+    private static final Logger LOGGER = Logger.getLogger(GlobalTracer.class.getName());
+
+    private static final GlobalTracer INSTANCE = new GlobalTracer();
+
+    /**
+     * The resolved {@link Tracer} to delegate the global tracing implementation to.<br>
+     * This can be either an {@link #register(Tracer) explicitly registered delegate}
+     * or the automatically resolved Tracer implementation.<br>
+     * Management of this reference is the responsibility of the {@link #tracer()} method.
+     */
+    private final AtomicReference<Tracer> globalTracer = new AtomicReference<Tracer>();
+
+    /**
+     * Private constructor to prevent outside instantiation of this singleton class.
+     */
+    private GlobalTracer() {
+    }
+
+    private Tracer getOrInitTracer() {
+        Tracer instance = globalTracer.get();
+        if (instance == null) {
+            final Tracer singleton = loadSingleton();
+            while (instance == null && singleton != null) {
+                globalTracer.compareAndSet(null, singleton);
+                instance = globalTracer.get();
+            }
+            LOGGER.log(Level.INFO, "Using GlobalTracer implementation: {0}.", instance);
+        }
+        LOGGER.log(Level.FINEST, "GlobalTracer: {0}.", instance);
+        return instance;
+    }
+
+    /**
+     * Explicit registration of a configured {@link Tracer} to back the behaviour
+     * of the {@link #tracer() global tracer}.
+     * <p>
+     * The previous global tracer is returned so it can be restored if necessary.
+     *
+     * @param delegate Tracer to delegate the tracing implementation to.
+     * @return The previous global tracer.
+     */
+    public static Tracer register(final Tracer delegate) {
```

Comment:
register makes me think multiple things are possible. a word like set, or replace is more explicit about what is going on here